### PR TITLE
Balance updates BE and FE

### PR DESCRIPTION
**Pull request by svitaly**
Created route to change balance of a specified user.
Balance can be topped up by going through checkout (buy new credits) and leaving review for others code (need an update here)
Now to upload new code u need 1 credit, it will be withdrawn after user creates code.

### CODE REVIEW

File: client/src/components/UploadCode.js [hunk 1]
```diff
@@ -37,6 +38,7 @@ export default function AddCodeDialog({ dispatch, open, handleClose }) {
     };
 
     //SEND REQUEST AND close dialog
+    updateBalance(state.user.id, -codeReviewPrice, dispatch);
```
Comment (Affeeq) [inline client/src/components/UploadCode.js:41]: there is a **-** before codeReviewPrice. not sure if that is intended

File: server/controllers/user.js [hunk 2]
```diff
@@ -91,6 +91,22 @@ module.exports = {
...
+  async updateBalance(req, res) {
+    try {
+      const userId = req.body.userId,
+        credits = req.body.credits;
+      const user = await User.findById(userId);
+      await User.findOneAndUpdate(
+        { _id: userId },
+        { balance: user.balance + credits }
+      );
+      const newUser = await User.findById(userId);
```
Comment (Affeeq) [inline server/controllers/user.js:103]: no problem doing this but can also simplify using {new:true} on findByIdAndUpdate to get the most updated user.

File: client/src/components/UploadCode.js (hunk 1 above)
Comment (svitaly) [reply client/src/components/UploadCode.js:41]: There is only one function to change balance on BE. If you want to withdraw credits you specify summ with a '-' minus sign

File: server/controllers/user.js (hunk 2 above)
Comment (svitaly) [reply server/controllers/user.js:103]: Thanks! will rewrite

**Pull request merged**